### DISCUSSION TITLE
NGINX Script

### DISCUSSION THREAD

**Discussion by jubjubrsx** [tteck/Proxmox#3403] [Help Needed]
First off.. 100% appreciate all the scripts love them and they work great...

although installing nginx tonight it does complain about an old version of npm and halts the install, and there is a bunch of warnings popping up.

**Comment by Mellowlynx** [comment]
Do you mean that it asks if you want to install the old version? The default is No, and for me, it works fine. 

```
    _   __      _               ____                           __  ___                                 
   / | / /___ _(_)___  _  __   / __ \_________ __  ____  __   /  |/  /___ _____  ____ _____ ____  _____
  /  |/ / __  / / __ \| |/_/  / /_/ / ___/ __ \| |/_/ / / /  / /|_/ / __  / __ \/ __  / __  / _ \/ ___/
 / /|  / /_/ / / / / />  <   / ____/ /  / /_/ />  </ /_/ /  / /  / / /_/ / / / / /_/ / /_/ /  __/ /    
/_/ |_/\__, /_/_/ /_/_/|_|  /_/   /_/   \____/_/|_|\__, /  /_/  /_/\__,_/_/ /_/\__,_/\__, /\___/_/     
      /____/                                      /____/                            /____/             
 
Using Default Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Using Container ID: 200
Using Hostname: nginxproxymanager
Using Disk Size: 4GB
Allocated Cores 2
Allocated Ram 1024
Using Bridge: vmbr0
Using Static IP Address: dhcp
Using Gateway IP Address: Default
Using Apt-Cacher IP Address: Default
Disable IPv6: No
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server Address: Host
Using MAC Address: Default
Using VLAN Tag: Default
Enable Root SSH Access: No
Enable Verbose Mode: No
Creating a Nginx Proxy Manager LXC using the above default settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 200 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 10.10.10.145 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.121.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed Python Dependencies
 ✓ Installed Openresty
 ✓ Installed Node.js
 ✓ Installed pnpm
Would you like to install an older version (v2.10.4)? <y/N> n
 ✓ Downloaded Nginx Proxy Manager v2.11.3
 ✓ Set up Enviroment
 ✓ Built Frontend
 ✓ Initialized Backend
 ✓ Created Service
 ✓ Customized Container
 ✓ Started Services
 ✓ Cleaned
 ✓ Set Container to Normal Resources
 ✓ Completed Successfully!

Nginx Proxy Manager should be reachable by going to the following URL.
         http://10.10.10.145:81
```

**Reply by jubjubrsx** [reply]
Well it works yes, didn't get me wrong... If you install in verbose mode you will see more...

**Reply by Mellowlynx** [reply]
That and the output you did not share, so guessing game for us then.
I will check it with the verbose mode later.

**Reply by jubjubrsx** [reply]
my bad,  most of the stuff I noticed was like

 `[36m-[0m WARN  deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1

 [36m\[0m WARN  26 deprecated subdependencies found: @npmcli/move-file@1.1.2, @npmcli/move-file@2.0.1, are-we-there-yet@3.0.1, chokidar@2.1.8, copy-concurrently@1.0.5, core-js@1.2.7, core-js@2.6.12, figgy-pudding@3.5.2, fs-write-stream-atomic@1.0.10, fsevents@1.2.13, gauge@4.0.4, glob@5.0.15, glob@7.1.7, glob@7.2.3, glob@8.1.0, inflight@1.0.6, move-concurrently@1.0.1, npmlog@6.0.2, q@1.5.1, resolve-url@0.2.1, rimraf@2.7.1, rimraf@3.0.2, source-map-resolve@0.5.3, source-map-url@0.4.1, urix@0.1.0, uuid@3.4.0

 [36m|[0mAlready up to date
Progress: resolved 906, reused 904, downloaded 0, added 0, done

 [36m/[0m
 WARN  Issues with peer dependencies found
.
└─┬ marionette.templatecache 1.0.0
  ├── ✕ unmet peer backbone@~1.3.3: found 1.6.0
  ├── ✕ unmet peer underscore@~1.8.3: found 1.13.6
  └─┬ backbone.marionette 4.0.0-beta.1
    ├── ✕ unmet peer backbone@~1.3.3: found 1.6.0
    └── ✕ unmet peer underscore@">= 1.8.3 <= 1.9.x": found 1.13.6
`
`WARNING in unable to locate 'app-images' at '/app/frontend/app-images'

WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets: 
  assets/tabler-ui/164feaa1f07e28af064ba4f8d1d8c930.svg (423 KiB)
  css/main.css (299 KiB)
  js/main.bundle.js?v=2.11.3 (608 KiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  main (906 KiB)
      css/main.css
      js/main.bundle.js?v=2.11.3

Child HtmlWebpackCompiler:
     2 assets
    Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
    Entrypoint HtmlWebpackPlugin_1 = __child-HtmlWebpackPlugin_1
    [0] ./node_modules/.pnpm/ejs-webpack-loader@2.2.2/node_modules/ejs-webpack-loader!./html/index.ejs 6.52 KiB {0} [built]
    [1] ./node_modules/.pnpm/ejs-webpack-loader@2.2.2/node_modules/ejs-webpack-loader!./html/login.ejs 6.69 KiB {1} [built]
Child mini-css-extract-plugin node_modules/.pnpm/css-loader@3.6.0_webpack@4.47.0/node_modules/css-loader/dist/cjs.js!node_modules/.pnpm/sass-loader@10.5.2_node-sass@9.0.0_webpack@4.47.0/node_modules/sass-loader/dist/cjs.js!scss/styles.scss:
    Entrypoint mini-css-extract-plugin = *
      [0] ./node_modules/.pnpm/css-loader@3.6.0_webpack@4.47.0/node_modules/css-loader/dist/cjs.js!./node_modules/.pnpm/sass-loader@10.5.2_node-sass@9.0.0_webpack@4.47.0/node_modules/sass-loader/dist/cjs.js!./scss/styles.scss 368 KiB {0} [built]
    [388] ./fonts/source-sans-pro/source-sans-pro-v14-latin-ext_latin-regular.woff2 100 bytes {0} [built]
    [389] ./fonts/source-sans-pro/source-sans-pro-v14-latin-ext_latin-regular.woff 99 bytes {0} [built]
    [390] ./fonts/source-sans-pro/source-sans-pro-v14-latin-ext_latin-italic.woff2 99 bytes {0} [built]
    [391] ./fonts/source-sans-pro/source-sans-pro-v14-latin-ext_latin-italic.woff 98 bytes {0} [built]
    [392] ./fonts/source-sans-pro/source-sans-pro-v14-latin-ext_latin-700italic.woff2 102 bytes {0} [built]
    [393] ./fonts/source-sans-pro/source-sans-pro-v14-latin-ext_latin-700italic.woff 101 bytes {0} [built]
    [394] ./fonts/source-sans-pro/source-sans-pro-v14-latin-ext_latin-700.woff2 96 bytes {0} [built]
    [395] ./fonts/source-sans-pro/source-sans-pro-v14-latin-ext_latin-700.woff 95 bytes {0} [built]
        + 387 hidden modules

 [36m|[0m[?25h
[K [1;92m✓[m [1;92mBuilt Frontend[m
   [33mInitializing Backend   [?25l
 [36m/[0m
 [36m-[0m
 [36m\[0m
 [36m|[0m
 [36m/[0mProgress: resolved 1, reused 0, downloaded 0, added 0

 [36m-[0m
 [36m\[0m WARN  deprecated json-schema-ref-parser@8.0.0: Please switch to @apidevtools/json-schema-ref-parser

 [36m|[0m
 [36m/[0m
 [36m-[0m
 [36m\[0m
 [36m|[0m
 [36m/[0m
 [36m-[0m
 [36m\[0mProgress: resolved 100, reused 32, downloaded 60, added 0

 [36m|[0m
 [36m/[0m
 [36m-[0m
 [36m\[0m
 [36m|[0m
 [36m/[0m
 [36m-[0m
 [36m\[0m
 [36m|[0m
 [36m/[0mProgress: resolved 243, reused 83, downloaded 144, added 0

 [36m-[0m
 [36m\[0m
 [36m|[0m
 [36m/[0m
 [36m-[0m
 [36m\[0m
 [36m|[0m
 [36m/[0m
 [36m-[0m WARN  14 deprecated subdependencies found: @humanwhocodes/config-array@0.11.14, @humanwhocodes/object-schema@2.0.3, @npmcli/move-file@1.1.2, are-we-there-yet@2.0.0, are-we-there-yet@3.0.1, gauge@3.0.2, gauge@4.0.4, glob@7.2.3, inflight@1.0.6, npmlog@5.0.1, npmlog@6.0.2, querystring@0.2.0, rimraf@3.0.2, uuid@3.4.0
Packages: +415
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
`

it might be nothing, just first time I saw WARN and depreciated subdependencies, and it went by so fast couldnt see it but thought I'd report it.

**Reply by Mellowlynx** [reply]
A deprecated warning like this is fine.
```
Pr WARN  deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
```
It's just letting you know, you are using an older version, and it might not work in the future and using is not really recommended anymore.
But you can use it, and it will give you time to make the changed needed to update.

You see this allot too in PHP, older WordPress plugins that run on PHP 8 or 8.1 will give a ton of these errors, the function is still in this PHP version, but in newer once it will be removed. As long as you use the older version, it should work, but isn't recommended and might break.

So while not ideal, it's fine for now.
